.travis.yml

```diff
@@ -0,0 +1,25 @@
+sudo: false
+language: python
+
+matrix:
+  include:
+     - python: 2.7
+     - python: 3.4
+       allow_failures: true
+     - python: 3.5
+       allow_failures: true
+     - python: 3.6
+       allow_failures: true
+     - python: 3.7
+       allow_failures: true
+       dist: xenial
+     - python: 3.8-dev
+       allow_failures: true
+       dist: xenial
+     - python: pypy
+       allow_failures: true
+     - python: pypy3.5
+       allow_failures: true
```

Comment:
It doesn't look like Travis CI honors these.

The documented syntax is [quite different](https://docs.travis-ci.com/user/customizing-the-build#rows-that-are-allowed-to-fail).

I would also suggest using a global `dist: xenial` and listing Python versions at the top level, because apparently allow_failures [doesn't work for jobs defined with `matrix.include` or something](https://docs.travis-ci.com/user/customizing-the-build#matching-jobs-with-allow_failures)?  The docs are not very clearly written.

Note that in that case you'll have to refer to pypy/pypy3.5 using fully-qualified versions such as pypy.2.7-6.0 and pypy3.5-6.0 because Travis CI is *gesticulates wildly bereft of speech*.